requirements/specifications/openrpc-extensions/app-passthrough-apis.md

```diff
@@ -0,0 +1,533 @@
+When an app calls a platform method with an `x-provided-by` extension, the platform **MUST** return an unavailable error if there is no [candidate app](#7-provider-candidates) to execute the provider method.
+
+```json
+{
+    "id": 1,
+    "error": {
+        "code": -50300,
+        "message": "Capability is unavailable."
+    }
+}
+```
+
+**TODO**: formalize our messages and codes.
+
+### 4.2. Direct pass-through
+A direct pass-through is where a single app provides a single response to a single request by another app.
+
+This section only applies to app provider methods that do not have an `event` tag and do not have the `x-multiple-providers` extension set to `true`.
+
+The platform method result schema **MUST** either:
+
+> Match the `x-response` schema on the provider method so that the result can be passed through.
+>
+> or
+> 
+> Have a property that matches the `x-response` schema on the provider method so that the result can
+> be composed and passed through.
+
+The platform **MUST** call the provider method from the [best candidate](#8-best-candidate) app and acquire the result.
+
+If the platform method result schema matches the `x-response` schema on the provider method then the value **MUST** be used as-is.
+
+Otherwise if the platform method result schema has a property that matches the `x-response` schema on the provider method then the value **MUST** be composed into an object under the corresponding property name and the platform **MUST** apply any [result transformations](#9-result-transformations) to the composed result.
+
+### 4.3. Aggregated pass-through results
+An aggregated pass-through is where many apps provide responses to a single request by another app. The results are aggregated inside of an array.
+
+This section only applies to app provider methods that do not have an `event` tag and do have the `x-multiple-providers` extension set to `true`.
+
+The platform method result schema **MUST** have a type of `array`.
+
+The platform method result schema **MUST** have an `items` sub-schema that either:
+
+> Matches the `x-response` schema on the provider method so that the result can be added to the final array.
+>
+> or
+>
+> Has a property that matches the `x-response` schema on the provider method so that the result can be composed
+> and added to the final array.
+
+The platform **MUST** call the provider method from each [candidate app](#7-provider-candidates) and aggregated all of the results into an array.
+
+If the platform method result `items` schema matches the `x-response` schema on the provider method then each provier value **MUST** be used as-is.
+
+Otherwise if the platform method result `items` schema has a property that matches the `x-response` schema on the provider method then each provider value **MUST** be composed into an object under the corresponding property name and the platform **MUST** apply any [result transformations](#9-result-transformations) to the composed result.
+
```

Comment:
Probably need a blurb here about provider timeouts.
If one provider returns and one times out, do we return an error or an array with one value.
(Assuming it would be latter)

Reply:
good catch. yeah, timeout is per-provider and we return the successful ones